torch/utils/model_dump/code.js

```diff
@@ -439,6 +439,114 @@ class OnePickleSection extends Component {
   }
 }
 
+function computeTensorMemory(numel, dtype) {
+  const sizes = {
+    "Byte": 1,
+    "Char": 1,
+    "Short": 2,
+    "Int": 4,
+    "Long": 8,
+    "Half": 2,
+    "Float": 4,
+    "Double": 8,
+    "ComplexHalf": 4,
+    "ComplexFloat": 8,
+    "ComplexDouble": 16,
+    "Bool": 1,
+    "QInt8": 1,
+    "QUInt8": 1,
+    "QInt32": 4,
+    "BFloat16": 2,
+  };
+  let dtsize = sizes[dtype];
+  if (!dtsize) {
+    throw new Error("Unrecognized dtype: " + dtype);
+  }
```

Comment:
I always found it interesting that one can look up a non-existent entry in a javascript object and it returns something falsy. I think there was a trick to identifying if it is falsy because it exists or because it doesn't exist, but I forget. Not relevant here.

Reply:
`dtsize === undefined` is pretty close, but I think `sizes.hasOwnProperty` is best.  Although if you want to get into that level of rigor with JS collections, you're probably better off using a Map.